Data.m

```diff
@@ -0,0 +1,89 @@
+classdef Data
+    %DATA this class loads and holds the IMU data
+    %   This class wraps the loading and handling of the IMU data
+
+    properties
+        IMU_data
+        windowSize
+        windowInterval
+    end
+
+    methods
+        function obj = Data(pattern, windowSize, interval)
+            arguments
+                pattern string = "IMU_Data_for_assignment/**/*.dat";
+                windowSize int16 = 1;
+                interval int16 = 1;
+            end
+            %Data Construct an instance of this class
+            
+            obj.windowInterval = interval;
+            obj.windowSize = windowSize;
+
+            files = dir(pattern);
+            columns = [];
+            for index = 1:length(files)
+                file = files(index);
+                path = file.folder + "\" + file.name;
+                data = obj.readDataFile(path);
+                obj.IMU_data{index} = data;
+                columns = [columns convertCharsToStrings(data.Properties.VariableNames)];
+            end
+            obj.IMU_data = obj.IMU_data;
+            columns = unique(columns);
+            for index = 1:length(obj.IMU_data)
+                for column = columns
+                    if ~any(strcmp(obj.IMU_data{index}.Properties.VariableNames, column)) 
+                        obj.IMU_data{index}.(column) = NaN(size(obj.IMU_data{index}, 1), 1);
+                    end
+                end
+                obj.IMU_data{index} = obj.slidingWindows(interval, obj.IMU_data{index});
+            end
+            obj.IMU_data = vertcat(obj.IMU_data{:});
```

Comment:
should only write to this at the end, in between should use better named variables